build: fix mailmap command

yexiang-aws opened the pull request:
Fix the way our debian packaging script uses Git mailmap option

before:
```
~/workspace/aws-ofi-nccl/master$ ./contrib/scripts/generate_source_packages.sh v1.14.1
fatal: unknown field name: mailmap
error: line 15: Empty tag: Version:
dpkg-source: warning: aws-ofi-nccl-/debian/changelog(l0): found end of file where expected first heading
dpkg-source: error: fatal error occurred while parsing aws-ofi-nccl-/debian/changelog
mv: cannot stat '/tmp/tmp.QCCav1IOtr/*.dsc': No such file or directory
mv: cannot stat '/tmp/tmp.QCCav1IOtr/*.debian.tar.xz': No such file or directory
mv: cannot stat '/tmp/tmp.QCCav1IOtr/*.src.rpm': No such file or directory
```

after
```
~/workspace/aws-ofi-nccl/master$ ./contrib/scripts/generate_source_packages.sh v1.14.1
Wrote: /tmp/tmp.U7x0xVApxY/libnccl-ofi-1.14.1-1.src.rpm
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: building aws-ofi-nccl using existing ./aws-ofi-nccl_1.14.1.orig.tar.gz
dpkg-source: info: building aws-ofi-nccl in aws-ofi-nccl_1.14.1-1.debian.tar.xz
dpkg-source: info: building aws-ofi-nccl in aws-ofi-nccl_1.14.1-1.dsc
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

rajachan reviewed (dismissed):
How was this working earlier? Was a minimum version of git required for this option?

rauteric commented on contrib/scripts/generate_debian_changelog.sh:
```diff
@@ -2,11 +2,11 @@
 tagname=%(refname:short) \
-tagger_name=%(taggername:mailmap) \
-tagger_email=%(taggeremail:mailmap) \
+tagger_name=%(taggername) \
```
I thought we added this to use the mailmap feature (https://github.com/aws/aws-ofi-nccl/blob/master/.mailmap) (to map email addresses to people). Do we know why it doesn't work?

yexiang-aws replied:
the support of `:mailmap ` usage in `git-for-each-ref` starts from git 2.43.0 - https://git-scm.com/docs/git-for-each-ref/2.43.0 .   The git in our Ubuntu cluster  `apt 2.4.13 (amd64)` is only  `2.34.1`

```
git is already the newest version (1:2.34.1-1ubuntu1.12).
```

bwbarrett reviewed (changes requested):
What changed and why and what's the feature change this implies?

yexiang-aws commented:
> What changed and why and what's the feature change this implies?

the support of `:mailmap ` usage in `git-for-each-ref` starts from git 2.43.0 - https://git-scm.com/docs/git-for-each-ref/2.43.0 .   The latest git in our Ubuntu cluster  `apt 2.4.13 (amd64)` is only  `2.34.1`

```
git is already the newest version (1:2.34.1-1ubuntu1.12).
```

Not sure when we introduced this script, if it's tested with the git version available on our Ubuntu dev clusters. But currently it's broken.  This change is only internal - only removes the mailmap use in debian changelog generation script and thus fixes the `generate_source_packages.sh` script to create debian tarball in our release process.  The `.mailmap` file is untouched and still works in git log messages to map contributors email/name to an unified name. But again, only if git version > 2.43.0 present.


Updated these details on PR description as well

bwbarrett commented:
you're missing the point.  That flag does something.  What does it do, why is not doing that ok?

yexiang-aws commented:
> you're missing the point. That flag does something. What does it do, why is not doing that ok?

the flag is only intended to map the tagger/committer names to a more formal / legal name in a configured manner.  So that as user move companies or change names in git configuration, it's automatically recognized as the same person.

Omitting the name mapping does not degrade our plugin software or break our release operation posture


```
fmt=" \
tagname=%(refname:short) \
tagger_name=%(taggername:mailmap) \
tagger_email=%(taggeremail:mailmap) \
tagger_when=%(taggerdate) \
committer_name=%(committername:mailmap) \
committer_email=%(committeremail:mailmap) \
committer_when=%(committerdate) \
"


git for-each-ref --shell --sort=-v:refname --format "$fmt" --merged "$tag" | {
```

bwbarrett commented:
But it means the author that shows up in the debian change log is different than what shows up in "git log"?

yexiang-aws commented:
> But it means the author that shows up in the debian change log is different than what shows up in "git log"?

it depends, for aws ofi nccl developers on dev cluster with old git version as default, the mailmap feature is not applied.  for someone with latest git, it can be different.  But currently in our case, we don't have any author in .mailmap configured with multiple/very different name mapping.  
i.e.  realistically speaking, author shows up with same name with the config as of now.

Without this removal, soon the next person works on 1.15.0 need to manually hack it again, or he uses a supported git version. 

But definitely, as soon as our dev cluster pulls a later git version, we should revert this change to keep names consistent.

bwbarrett commented:
>it depends, for aws ofi nccl developers on dev cluster with old git version as default, the mailmap feature is not applied. for someone with latest git, it can be different. But currently in our case, we don't have any author in .mailmap configured with multiple/very different name mapping.

.mailmap begs to differ:
```
<akkart@amazon.com> <114260793+arunkarthik-akkart@users.noreply.github.com>
<bbarrett@amazon.com> <ubuntu@ip-10-192-10-147.ec2.internal>
<eraut@amazon.com> <10216922+rauteric@users.noreply.github.com>
<eraut@amazon.com> <rauteric@users.noreply.github.com>
<hunnorth@amazon.com> <119898856+hunnorth@users.noreply.github.com>
<raghunch@amazon.com> <craghun@amazon.com>
<raghunch@amazon.com> <rajachan@users.noreply.github.com>
<rashika@amazon.com> <42682819+rashikakheria@users.noreply.github.com>
<rashika@amazon.com> <rashika@f4d4888cbd16.ant.amazon.com>
```

yexiang-aws commented:
PR CI g4dn tcp test failure due to OOB

```
262144         65536     float     sum      -1   1163.7  0.225273  0.422387      0   1306.2  0.200691  0.376296     16

# Out of bounds values : 16 FAILED
# Avg bus bandwidth    : 0.507692 
```

yexiang-aws commented:
bot:aws:retest
